src/main/java/Main.java

```diff
@@ -0,0 +1,47 @@
+import db.FileWriter;
+import db.FileWriterImpl;
+import db.TransactionReader;
+import db.TransactionReaderImpl;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import service.FruitService;
+import service.FruitServiceImpl;
+import service.ReportService;
+import service.ReportServiceImpl;
+import strategy.OperationStrategy;
+import strategy.OperationStrategyImpl;
+import strategy.activities.BalanceHandler;
+import strategy.activities.OperationHandler;
+import strategy.activities.PurchaseHandler;
+import strategy.activities.ReturnHandler;
+import strategy.activities.SupplyHandler;
+
+public class Main {
+    private static final String filePath = "src/main/resources/file.csv"; // data from file
+    private static final TransactionReader readFile = new TransactionReaderImpl();
+    private static final FileWriter fileWriter = new FileWriterImpl();
+    private static final ReportService reportService = new ReportServiceImpl();
+
+    public static void main(String[] args) {
+        // read and convert data
+        List<FruitTransaction> dataFromCsv = readFile.readTransactionsFromFile(filePath);
+
+        // process
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap = Map.of(
+                FruitTransaction.Operation.BALANCE, new BalanceHandler(),
+                FruitTransaction.Operation.SUPPLY, new SupplyHandler(),
+                FruitTransaction.Operation.PURCHASE, new PurchaseHandler(),
+                FruitTransaction.Operation.RETURN, new ReturnHandler()
+        );
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+
+        FruitService fruitService = new FruitServiceImpl(operationStrategy);
+
+        // create report
+        String report = reportService.generateReport(fruitService.processData(dataFromCsv));
```

Comment:
processData should write in db while generateReport should read from it
```suggestion
        fruitService.processData(dataFromCsv);

        // create report
        String report = reportService.generateReport();
```